openstudiocore/src/utilities/core/FilesystemHelpers.cpp

```diff
@@ -172,5 +187,80 @@ namespace openstudio {
     {
       return to_time_t(openstudio::filesystem::last_write_time(t_path));
     }
+
+    openstudio::path create_temporary_directory(const openstudio::path &basename)
+    {
+      // making this count static/atomic so that we reduce the chance of collisions
+      // on each run of the binary. This is threadsafe, with the atomic
+      static std::atomic<unsigned int> count = 0;
+      constexpr auto allowed_attempts = 1000;
+
+      const auto temp_dir = openstudio::filesystem::temp_directory_path();
+
+      int attempts{0};
+
+      while (attempts < allowed_attempts) {
+        // concat number to path basename, without adding a new path element
+        auto filename = basename;
+        filename += openstudio::toPath("-" + std::to_string(count++));
```

Comment:
@lefticus if these never get cleaned up then this will start to always fail right?

Reply:
Yeah, there might be a better way to do this, I'll spend some more time on it.

Reply:
Updated to include a timestamp to further reduce possible issues. e19550